pkg/controller/openpitrix/helmapplication/helm_application_controller.go

```diff
@@ -181,3 +188,82 @@ func (r *ReconcileHelmApplication) SetupWithManager(mgr ctrl.Manager) error {
 func inAppStore(app *v1alpha1.HelmApplication) bool {
 	return strings.HasSuffix(app.Name, v1alpha1.HelmApplicationAppStoreSuffix)
 }
+
+// cleanupDangingApp delete the app when it is not active and not suspended,
+// sets the workspace label to empty and remove parts of the appversion when app state are active or suspended
+func (r *ReconcileHelmApplication) cleanupDangingApp(ctx context.Context, app *v1alpha1.HelmApplication) error {
+	if app.Annotations[constants.DangingAppCleanupKey] == constants.CleanupDangingAppOngoing {
+		// Just delete the app when the state is not active or not suspended.
+		if app.Status.State != v1alpha1.StateActive && app.Status.State != v1alpha1.StateSuspended {
```

Comment:
What if app.Status is nil?

Reply:
Status is a struct, not a pointer, it can not be nil.